agent/stats/container.go

```diff
@@ -19,23 +19,19 @@ import (
 	"fmt"
 	"time"
 
+	"github.com/aws/amazon-ecs-agent/agent/config"
 	"github.com/aws/amazon-ecs-agent/agent/dockerclient"
 	"github.com/aws/amazon-ecs-agent/agent/dockerclient/dockerapi"
 	"github.com/aws/amazon-ecs-agent/agent/stats/resolver"
 	"github.com/aws/amazon-ecs-agent/agent/utils/retry"
 	"github.com/cihub/seelog"
 )
 
-const (
-	// SleepBetweenUsageDataCollection is the sleep duration between collecting usage data for a container.
-	SleepBetweenUsageDataCollection = 1 * time.Second
+// default size of stats queue, assumes 1s (streaming) collection interval
+// (seconds in one publish interval) / 1 (1s interval) * 4 (intervals)
+var queueSize int = int(config.DefaultContainerMetricsPublishInterval.Seconds()) / 1 * 4
```

Comment:
I realize that go will follow order of precedence and this will end up being metricsPublishInterval / 4

It's just odd to see the `/ 1` -- I'd prefer to see these be named, aka take what you've got in the comment above and just call these something like `publishIntervalDuration := 1` and `publishIntervalCount := 1`.

Reply:
yeah this is mostly just written out so it's clear that it's following the same formula as in the StartStatsCollection function below. That being said, this doesn't actually need to be a package-level var for any reason. I will change it to a local variable in this function and I think that will make it more clear that this is for streaming stats only too.

Reply:
local var is perfect.